### PR TITLE
[connectivity] Fix IllegalArgumentException & Add Log warning 

### PR DESCRIPTION
**Pull request by hamdikahloun**
## Description

* Android: Add CHANGE_WIFI_STATE permission.
* Android: Add Log warning for unsatisfied requirement(s) in Android P or higher.
* Android: Fix IllegalArgumentException.
* Android: Update Example project.

## Related Issues
https://github.com/flutter/flutter/issues/68265

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/connectivity/connectivity/android/src/main/java/io/flutter/plugins/connectivity/Connectivity.java [hunk 1]
```diff
@@ -103,4 +118,38 @@ private String getNetworkTypeLegacy() {
   public ConnectivityManager getConnectivityManager() {
     return connectivityManager;
   }
+
+  private Boolean checkRequirement() {
```
Comment (dnfield) [inline packages/connectivity/connectivity/android/src/main/java/io/flutter/plugins/connectivity/Connectivity.java:122]: nit: Can we name this more descriptively? Maybe something like `checkPermissions`?

File: packages/connectivity/connectivity/android/src/main/java/io/flutter/plugins/connectivity/Connectivity.java [hunk 2]
```diff
@@ -103,4 +118,38 @@ private String getNetworkTypeLegacy() {
...
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) return true;
```
Comment (dnfield) [inline packages/connectivity/connectivity/android/src/main/java/io/flutter/plugins/connectivity/Connectivity.java:123]: nit: avoid having `return` on the same line, and use `{}`
```suggestion
    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
      return true;
    }
```

**Review by dnfield — APPROVED** [review]
LGTM modulo some nits

File: packages/connectivity/connectivity/android/src/main/java/io/flutter/plugins/connectivity/Connectivity.java (hunk 1 above)
Comment (hamdikahloun) [reply packages/connectivity/connectivity/android/src/main/java/io/flutter/plugins/connectivity/Connectivity.java]: PR updated. Thank you

**Review by dnfield — APPROVED** [review]
LGTM when tree is green.

**Comment by hamdikahloun** [comment]
cc @dnfield 
closed . 
the code will be added in wifi_info_flutter.

**Pull request closed without merging**